src/mistral-client.ts

```diff
@@ -147,90 +136,26 @@ class MistralClient {
             `Response: ${await response.text()}`,
             `Attempt: ${attempts + 1}`,
           );
-          // eslint-disable-next-line max-len
-          await new Promise((resolve) =>
-            setTimeout(resolve, Math.pow(2, (attempts + 1)) * 500),
-          );
+          await new Promise((resolve) => setTimeout(resolve, Math.pow(2, attempts + 1) * 500));
         } else {
-          throw new MistralAPIError(
-            `HTTP error! status: ${response.status} ` +
-            `Response: \n${await response.text()}`,
-          );
+          throw new MistralAPIError(`HTTP error! status: ${response.status}`);
         }
-      } catch (error) {
+      } catch (error: any) {
         console.error(`Request failed: ${error.message}`);
-        if (error.name === 'MistralAPIError') {
-          throw error;
-        }
         if (attempts === this.maxRetries - 1) throw error;
-        // eslint-disable-next-line max-len
-        await new Promise((resolve) =>
-          setTimeout(resolve, Math.pow(2, (attempts + 1)) * 500),
-        );
+        await new Promise((resolve) => setTimeout(resolve, Math.pow(2, attempts + 1) * 500));
       }
     }
     throw new Error('Max retries reached');
-  };
-
-  /**
-   * Creates a chat completion request
-   * @param {*} model
-   * @param {*} messages
-   * @param {*} tools
-   * @param {*} temperature
-   * @param {*} maxTokens
-   * @param {*} topP
-   * @param {*} randomSeed
-   * @param {*} stream
-   * @param {*} safeMode deprecated use safePrompt instead
-   * @param {*} safePrompt
-   * @param {*} toolChoice
-   * @param {*} responseFormat
-   * @return {Promise<Object>}
-   */
-  _makeChatCompletionRequest = function(
-    model,
-    messages,
-    tools,
-    temperature,
-    maxTokens,
-    topP,
-    randomSeed,
-    stream,
-    safeMode,
-    safePrompt,
-    toolChoice,
-    responseFormat,
-  ) {
-    // if modelDefault and model are undefined, throw an error
-    if (!model && !this.modelDefault) {
-      throw new MistralAPIError(
-        'You must provide a model name',
-      );
-    }
-    return {
-      model: model ?? this.modelDefault,
-      messages: messages,
-      tools: tools ?? undefined,
-      temperature: temperature ?? undefined,
-      max_tokens: maxTokens ?? undefined,
-      top_p: topP ?? undefined,
-      random_seed: randomSeed ?? undefined,
-      stream: stream ?? undefined,
-      safe_prompt: (safeMode || safePrompt) ?? undefined,
-      tool_choice: toolChoice ?? undefined,
-      response_format: responseFormat ?? undefined,
-    };
-  };
+  }
 
   /**
    * Returns a list of the available models
-   * @return {Promise<Object>}
+   * @returns 
```

Comment:
consistent return vs returns?

Personally I would just lose the JSDoc now that it's typescript

Reply:
due to this discussion:
https://github.com/mistralai/client-js/pull/73#discussion_r1598054622

I agree with jsdoc making code more readable, but okay with both.